packages/agents-manager/src/hooks/use-persisted-agent-state.ts

```diff
@@ -0,0 +1,269 @@
+	 * @default 'agents-manager-state'
+	 */
+	preferenceKey?: string;
+	/**
+	 * Function to save preferences to server
+	 * Should accept the preference key and value
+	 */
+	savePreference?: ( key: string, value: PersistedAgentState ) => Promise< void >;
+	/**
+	 * Function to load preferences from server
+	 * Should return the persisted state or null
+	 */
+	loadPreference?: ( key: string ) => Promise< PersistedAgentState | null >;
+	/**
+	 * Debounce delay for saving preferences (ms)
+	 * @default 1000
+	 */
+	debounceMs?: number;
+}
+
+export interface UsePersistedAgentStateResult {
+	/**
+	 * Current persisted state
+	 */
+	state: PersistedAgentState;
+	/**
+	 * Update session ID
+	 */
+	setSessionId: ( sessionId: string ) => void;
+	/**
+	 * Update open/closed state
+	 */
+	setIsOpen: ( isOpen: boolean ) => void;
+	/**
+	 * Update dock/undock state
+	 */
+	setIsDocked: ( isDocked: boolean ) => void;
+	/**
+	 * Update multiple state fields at once
+	 */
+	updateState: ( partial: Partial< PersistedAgentState > ) => void;
+	/**
+	 * Whether state is currently being loaded
+	 */
+	isLoading: boolean;
+}
+
+/**
+ * Default save function using wpcom-proxy-request or apiFetch
+ */
+const defaultSavePreference = async (
+	key: string,
+	value: PersistedAgentState
+): Promise< void > => {
+	// Try to use wpcom API if available
+	if ( typeof window !== 'undefined' && ( window as any ).wpcomRequest ) {
+		const wpcomRequest = ( window as any ).wpcomRequest;
+		try {
+			await wpcomRequest( {
+				path: '/me/preferences',
+				apiNamespace: 'wpcom/v2',
+				method: 'PUT',
+				body: {
+					calypso_preferences: {
+						[ key ]: value,
+					},
+				},
+			} );
+			return;
+		} catch ( error ) {
+			// eslint-disable-next-line no-console
+			console.warn( '[usePersistedAgentState] Failed to save to wpcom:', error );
+		}
+	}
+
+	// Fallback to WordPress apiFetch if available
+	if ( typeof window !== 'undefined' && ( window as any ).wp?.apiFetch ) {
+		const apiFetch = ( window as any ).wp.apiFetch;
+		try {
+			await apiFetch( {
+				path: '/ai-agent/state',
+				method: 'PUT',
+				data: { [ key ]: value },
+			} );
+		} catch ( error ) {
+			// eslint-disable-next-line no-console
+			console.warn( '[usePersistedAgentState] Failed to save via apiFetch:', error );
+		}
+	}
+};
+
+/**
+ * Default load function
+ */
+const defaultLoadPreference = async ( key: string ): Promise< PersistedAgentState | null > => {
+	// Try to use wpcom API if available
+	if ( typeof window !== 'undefined' && ( window as any ).wpcomRequest ) {
+		const wpcomRequest = ( window as any ).wpcomRequest;
+		try {
+			const response = await wpcomRequest( {
+				path: '/me/preferences',
+				apiNamespace: 'wpcom/v2',
+				method: 'GET',
+			} );
+			return response?.calypso_preferences?.[ key ] || null;
+		} catch ( error ) {
+			// eslint-disable-next-line no-console
+			console.warn( '[usePersistedAgentState] Failed to load from wpcom:', error );
+		}
+	}
+
+	return null;
+};
```

Comment:
This won't work in atomic. You'll need an adapter in Jetpack.

For inspiration:

- Client: https://github.com/Automattic/wp-calypso/blob/trunk/packages/help-center/src/hooks/use-persisted-history.ts
- Jetpack: https://github.com/Automattic/jetpack/blob/trunk/projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php